### PR TITLE
modify escrow to support IOU escrow

### PR DESCRIPTION
**Pull request by luleigreat**

### CODE REVIEW

**Comment by mellery451** [comment]
FYI - build failures related to `dirDelete`:

```
In file included from /opt/jenkins/workspace/rippled-dev_PR-2672/src/ripple/unity/app_tx.cpp:33:
/opt/jenkins/workspace/rippled-dev_PR-2672-MSAUSDQC42TBL765KVFJX4JGAVGIIOYMUCPR3IAZ7OEGY2HMRDXQ/src/ripple/app/tx/impl/Escrow.cpp:570:19: error: use of undeclared identifier 'dirDelete'; did you mean 'delete'?
                TER const ter = dirDelete(ctx_.view(), true,
```

This might have happened if you worked off an earlier version of the code and then rebased.

**Comment by nbougalis** [comment]
@luleigreat: if you have any questions please don't hesitate to ask here! We're excited to see third party developers working on developing new `rippled` features and would be happy to help answer any questions.

File: src/ripple/app/tx/impl/Escrow.cpp [hunk 1]
```diff
@@ -198,34 +201,72 @@ EscrowCreate::doApply()
...
+		else
+		{
+			SLE::pointer sleRippleStateDst = view().peek(
+				keylet::line(dest, amount.getIssuer(), amount.getCurrency()));
+			if (!sleRippleStateDst)
+				return tecNO_LINE;
+			bool const bHigh = dest > amount.getIssuer();
+			auto limit = sleRippleStateDst->getFieldAmount(!bHigh ? sfLowLimit : sfHighLimit);
+			if (limit < amount)
+				return temBAD_PATH;
```
[the remark this answers is not in the document]
Comment (JoelKatz) [reply src/ripple/app/tx/impl/Escrow.cpp]: You also can't return a tem* error unless the transaction is in fact malformed. A tem* error means the transaction can't be successfully applied to any ledger (under the current rules). This also opens possible attack vectors as the transaction may appear valid (thereby getting relayed) and later appear invalid due to changes in the RippleState (thereby not claiming a fee). If you had to check the ledger, you want to claim a fee if possible, and that is possible here.

File: src/ripple/app/tx/impl/Escrow.cpp [hunk 2]
```diff
@@ -581,10 +702,54 @@ EscrowCancel::doApply()
...
+		ctx_.view().update(sled);
+	}
+	else if (account != amount.getIssuer())
+	{
+		SLE::pointer sled = view().peek(
+			keylet::line(account, amount.getIssuer(), amount.getCurrency()));
+		STAmount const& balance = (*sled)[sfBalance];
+		bool const bHigh = account > amount.getIssuer();
+		auto limit = sled->getFieldAmount(!bHigh ? sfLowLimit : sfHighLimit);
+		if (bHigh)
```
[the remark this answers is not in the document]
Comment (JoelKatz) [reply src/ripple/app/tx/impl/Escrow.cpp]: We similarly ignore limits when your offer is executing. Other actions can implicitly agree to accept an IOU and that is the case here.

**Comment by nbougalis** [comment]
@luleigreat, we appreciate your proposal. Unfortunately, it has not seen any activity for several months, so I am going to have to close it as this point.

I hope that you will revive it and resubmit it for consideration. If you need help, please don't hesitate to ask; plenty of other developers would be happy to work with you on this.

Good luck!

**Comment by GildedHonour** [comment]
Is help needed on this? Whom can I discuss it with?

**Comment by intelliot** [comment]
@GildedHonour Yes, feel free to discuss in #3233. Also, feel free to open a new PR.

**Pull request closed without merging**